content/html-css/blocks/what-is-css/index.md

```diff
@@ -0,0 +1,118 @@
+
+<details>
+<summary>What does precedence mean?</summary>
+
+It means hierarchy. One hierarchy in CSS is inheritance: children inherit from parents. If a parent node has red text, all its children will inherit that red text color. But when another rule comes after, and overrules, it takes precedence. Everything in CSS is determined by order.
+
+Order is so important in CSS because CSS is a declarative programming language that programs the layout of boxes.
+
+CSS renders the nodes in our DOM tree as nested boxes, and we program their appearance using fonts, colours, and spacing.
+
+</details>
+
+<details>
+<summary>What does declarative mean?</summary>
+
+It means, broadly speaking, that the entire code is processed and then applied, from top to bottom in one go. You don’t write for loops or if/else statements; there are no functions as such; you just _declare_ the outcome.
+
+In CSS you do the 'what', and the browser does the 'how'.
+
+</details>
+
+<details>
+<summary>What is specificity?</summary>
+
+Specificity is a [weighting system for CSS rules](https://specifishity.com/). The more specific your selector, the more power it has. For example: IDs can only ever mean _one_ node. IDs are unique, so they are _more specific_ than classes, which could mean many nodes.
+
+Classes added to html element are _more specific_ than just a plain element selector, as a class of `p` is _more specific_ than just all `p`. If you get two elements with the same specificity then the one that comes last in the order wins.
+
+The browser sorts all your rulesets, or declarations, into a defined order according to the power, or weight, of the declaration and applies the result, like a game of cards where higher value cards beat lower value cards.
+
+Stronger rules _override_ weaker rules. Everything is sorted and sorted until eventually a final value wins out and the view is rendered. The browser sorts like this:
+
+1. Match the media
+1. Sort by origin
+1. Sort by specificity
+1. Sort by document **order**
+
+</details>
+
+<details><summary>What do we mean by order?</summary>
+
+Order means 'what comes first?'.
+
+Another way to think of ‘earlier and later’ is ‘farther and nearer’. The _closer_ the declaration is to the node in the DOM, the ‘later’ it is in the CSS. So an inline style overrides a style set in a style block, which overrides a stylesheet linked in the head, which overrides any stylesheet linked higher up in the head, which all ultimately overrides the user agent style that comes in the browser.
+
+</details>
+
+<details>
+<summary>What does specify mean?</summary>
+
+Identify clearly. Name explicitly.
+
+</details>
+
+<details>
+<summary>What does render mean in context of a browser?</summary>
+
+Render means to hand over, actually - to give back. In the browser, it means to show the result of all these computations we have talked about. It turns the code into pixels and paints them on your screen. So it’s the end result - the final face.
```

Comment:
Not sure I know what "the final face" means?

Reply:
Hmm, good point. Let's merge this and then do another revision